### PR TITLE
Dashboard

### PR DESCRIPTION
**Pull request by ctlong**
- Added delete form option
- Style changes
- added validation for form names
- added navigation to editing page for existing forms

### CODE REVIEW

File: app/assets/javascripts/angular/controller/forms/listForms.js.erb [hunk 1]
```diff
@@ -20,10 +20,28 @@ app.controller('ListFormsCtrl', ['$scope', '$http','$auth','$location', function
...
     });
   }
 
+  $scope.goToForm = function(form) {
+    $location.path('forms/' + response.id + '/edit');
+  }
+
+  $scope.delete = function(form) {
+    if(confirm('This will delete all data associated with this form, and is irreversible')) {
+      $scope.forms.splice($scope.forms.indexOf(form),1);
```
Comment (h8rry) [inline app/assets/javascripts/angular/controller/forms/listForms.js.erb:36]: What are we trying to do here?

File: app/models/form.rb [hunk 2]
```diff
@@ -1,5 +1,17 @@
+class NameValidator < ActiveModel::Validator
```
Comment (h8rry) [inline app/models/form.rb:1]: Why make a new class when you can do inline validators?

Comment (h8rry) [inline app/models/form.rb:1]: http://guides.rubyonrails.org/active_record_validations.html#validation-helpers

File: app/assets/templates/forms/index.html [hunk 3]
```diff
@@ -3,14 +3,16 @@
...
+        <button class="btn btn-primary btn-hg" ng-click="createForm()">
           Create a form
         </button>
-        <div class="col-xs-4" ng-repeat="form in forms track by $index">
-          <h2>{{ form.name }}</h2>
-        </div>
-        <h4 ng-show="noForms">No Forms</h4>
+      </div>
+      <h4 ng-show="noForms">No Forms</h4>
+      <div class="form-link" ng-repeat="form in forms track by $index" ng-mouseover="showOptions=true" ng-mouseleave="showOptions=false">
```
Comment (h8rry) [inline app/assets/templates/forms/index.html:12]: good

File: app/models/form.rb (hunk 2 above)
Comment (ctlong) [inline app/models/form.rb:1]: I wanted to do it my way.

**Comment by ctlong** [comment]
Edited except for the validation. I didn't change it because the inline validations aren't as flexible as I would like.

File: app/models/form.rb (hunk 2 above)
Comment (h8rry) [inline app/models/form.rb:1]: ok

**Comment by h8rry** [comment]
Can merge yo

**Pull request merged**